File: app_java/src/main/java/com/example/app_java/PrinterViewModel.java
package com.example.app_java;

import android.content.Context;

import androidx.lifecycle.ViewModel;

import com.sunmi.printerx.PrinterSdk;
import com.sunmi.printerx.SdkException;

import java.util.List;

public class PrinterViewModel extends ViewModel {

    public void initPrinter(Context context) {

        try {
            PrinterSdk.getInstance().getPrinter(context, new PrinterSdk.PrinterListen() {

                @Override
                public void onDefPrinter(PrinterSdk.Printer printer) {
                    PrinterHolder.selectPrinter = printer;
                }

                @Override
                public void onPrinters(List<PrinterSdk.Printer> printers) {
                }
            });
        } catch (SdkException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/example/printersample/viewModel/PrinterViewModel.kt
package com.example.printersample.viewModel

import android.content.Context
import androidx.lifecycle.ViewModel
import com.example.printersample.utils.Constant
import com.sunmi.printerx.PrinterSdk

class PrinterViewModel : ViewModel() {

    fun initPrinter(context: Context) {
        PrinterSdk.getInstance().getPrinter(context, object : PrinterSdk.PrinterListen {

            override fun onDefPrinter(printer: PrinterSdk.Printer?) {
                Constant.selectPrinter = printer
            }

            override fun onPrinters(printers: MutableList<PrinterSdk.Printer>?) {
            }
        })
    }


}